### DISCUSSION TITLE
Why can I not call count with async?

### DISCUSSION THREAD

**Discussion by JohnLKramer** [prisma/prisma#7253] [Q&A] [answered]
Hey guys,

I'm following this tutorial:
https://www.howtographql.com/graphql-js/8-filtering-pagination-and-sorting/

And I just ran into this error:

```
/Users/jkramer6/source/examples/hackernews-node/src/resolvers/Query.js:17
  const count = await context.prisma.link.count({ where })
                ^^^^^

SyntaxError: await is only valid in async function
    at wrapSafe (internal/modules/cjs/loader.js:979:16)
    at Module._compile (internal/modules/cjs/loader.js:1027:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
```
If I look at the prisma interface in node_modules, I find this:

```
count<T extends LinkCountArgs>(
      args?: Subset<T, LinkCountArgs>,
    ): PrismaPromise<
      T extends _Record<'select', any>
        ? T['select'] extends true
          ? number
          : GetScalarType<T['select'], LinkCountAggregateOutputType>
        : number
    >
```

I would guess that PrsimaPromise is a promise, but maybe not.

It works without the async, but I'm curious what's going on.

Thanks!

**Comment by pantharshit00** [comment, marked as answer]
Did you mark your function async like so: 
```ts
   async function feed(parent, args, context, info) {
// ^^^^
// |------ await won't work without marking function async
```


> It works without the async, but I'm curious what's going on.

Actually, you can return a promise from a graphql resolver so that why it worked. See: https://graphql.org/learn/execution/#asynchronous-resolvers

**Reply by JohnLKramer** [reply]
That was it. Thank you!!!
